pyEpiabm/pyEpiabm/tests/test_unit/test_sweep/test_spatial_sweep.py

```diff
@@ -14,69 +13,147 @@ class TestSpatialSweep(TestMockedLogs):
 
-        # Check when all (one) nan in distance, won't call nan_to_num
-        mock_dist.return_value = 0
-        test_list = test_sweep.find_infectees(self.cell_inf, [cell_susc], 1)
-        self.assertFalse(mock_nan.called)
-        self.assertEqual(test_list, [infectee])
+        self.assertEqual(self.cell_inf.nearest_neighbours, {})
 
-        mock_dist.return_value = 2
-        test_list = test_sweep.find_infectees(self.cell_inf, [cell_susc], 1)
-        self.assertFalse(mock_nan.called)
-        self.assertEqual(test_list, [infectee])
+        Parameters.instance().infection_radius = 1000
+        test_pop = self.pop
+        test_sweep = SpatialSweep()
+        mock_dist.return_value = 2.2
+        self.assertEqual(test_sweep.bind_population(test_pop), None)
+        self.assertEqual(test_sweep.bind_population(test_pop), None)
 
-        mock_nan.return_value = [2, 0.0000001]
-        mock_dist.side_effect = [0, 2]
-        test_pop.add_cells(1)
-        third_cell = test_pop.cells[2]
-        third_cell.add_microcells(1)
-        third_cell.microcells[0].add_people(1)
+    @mock.patch("logging.exception")
+    @mock.patch("numpy.nan_to_num")
+    @mock.patch("pyEpiabm.utility.DistanceFunctions.dist_euclid")
+    def test_find_infectees_successful(self, mock_dist, mock_nan, mock_logger):
+        Parameters.instance().infection_radius = 1000
+        test_pop = self.pop
+        test_sweep = SpatialSweep()
+        mock_dist.return_value = 2.2
+        test_sweep.bind_population(test_pop)
+        test_list = test_sweep.\
+            find_infectees(self.cell_inf, [self.cell_susc], 1)
+        self.assertFalse(mock_nan.called)
+        self.assertEqual(test_list, [self.infectee])
 
-        test_list = test_sweep.find_infectees(self.cell_inf, [cell_susc,
-                                                              third_cell], 1)
-        mock_nan.assert_called_once_with([np.nan, 0.5], nan=0.5)
-        self.assertEqual(test_list, [infectee])
+    @mock.patch("logging.exception")
+    @mock.patch("numpy.nan_to_num")
+    @mock.patch("pyEpiabm.utility.DistanceFunctions.dist_euclid")
+    def test_find_infectees_fails(self, mock_dist, mock_nan, mock_logger):
+        Parameters.instance().infection_radius = 0.0001
+        # Assert a basic population
+        test_pop = self.pop
+        test_sweep = SpatialSweep()
+        mock_dist.return_value = 2.2
+        test_sweep.bind_population(test_pop)
 
         # test the assert that the distance weights has correct length
-        mock_dist.side_effect = [0, 2]
-        mock_nan.return_value = [1]
-        self.assertRaises(AssertionError, test_sweep.find_infectees,
-                          self.cell_inf, [cell_susc, third_cell], 1)
+        # mock_dist.side_effect = [0, 2]
+        # mock_nan.return_value = [1]
+        # self.assertRaises(AssertionError, test_sweep.find_infectees(
+        #                   self.cell_inf, [self.cell_susc], 1))
```

Comment:
Well the test doesn't seem to be needed but I'm not sure

Reply:
Why not? It looks to me but like it explicitly tests [L129](https://github.com/SABS-R3-Epidemiology/epiabm/blob/d314586c69e2dde2d6fbe70cc1b05ac2846b155b/pyEpiabm/pyEpiabm/sweep/spatial_sweep.py#L129), but am I missing something?

Reply:
L129 is not missed

Reply:
Just because something is included in coverage, doesn't mean that we are testing whether it works as we expect! In this case, we want to check the assertion error is thrown when we need it to be, so that's the role of the commented code. In general, getting 100% coverage is a good first step in testing but it doesn't mean we have actually tested everything - unit tests should check the functionality of each step in the code (not just check that it runs once without an error)

Reply:
Ok so this code was for an assert statement that is no longer included or needed in the code due to a changein how distance_weights is created, commented code is now deleted